pennylane/ops/op_math/exp.py

```diff
@@ -280,3 +288,153 @@ def simplify(self):
         if isinstance(self.base, qml.ops.op_math.SProd):  # pylint: disable=no-member
             return Exp(self.base.base.simplify(), self.coeff * self.base.scalar)
         return Exp(self.base.simplify(), self.coeff)
+
+    def generator(self):
+        r"""Generator of an operator that is in single-parameter-form.
+
+        For example, for operator
+
+        .. math::
+
+            U(\phi) = e^{i\phi (0.5 Y + Z\otimes X)}
+
+        we get the generator
+
+        >>> U.generator()
+          (0.5) [Y0]
+        + (1.0) [Z0 X1]
+
+        """
+        if np.real(self.coeff) != 0 or not self.base.is_hermitian:
+            raise GeneratorUndefinedError(
+                f"Exponential with coefficient {self.coeff} and base operator {self.base} does not have a generator."
+            )
+        return self.base
+
+    @property
```

Comment:
If this code is just duplicated from `Operation`, we could just make `Exp` inherit from `Operation` instead of `Operator` and simply inherit the code.

Reply:
It's identical to the code from Operation, but I wasn't sure how to make it inherit that without breaking things. Right now `Exp` inherits from `SymoblicOp` inherits from `Operator`. What would you suggest?

Reply:
Would it be okay to move it all the way up to Operator so both this and Operation can inherit it?